apps/web/components/OrderPropertyFileUpload/OrderPropertyFileUpload.vue

```diff
@@ -0,0 +1,134 @@
+            <div class="underline">
+              {{ $t('orderProperties.upload.uploadFile') }}
+            </div>
+          </div>
+        </SfButton>
+        <div>
+          <div class="mr-5">
+            <span> {{ $t('orderProperties.upload.acceptedFormats') }}: </span>
+            <span v-for="(suportedFormat, i) in suportedFormats" :key="suportedFormat" class="m-0 p-0">
+              <span>
+                {{ suportedFormat }}
+              </span>
+              <span v-if="i < suportedFormats.length - 1">,</span>
+            </span>
+          </div>
+          <span> {{ $t('orderProperties.upload.maximumFileSize') }}: 10mb </span>
+        </div>
+      </div>
+
+      <div v-if="hasTooltip" class="w-[28px]">
+        <slot name="tooltip" />
+      </div>
+    </div>
+
+    <div v-if="loading" class="uploadArea">
+      <div align="center" class="m-5">
+        <SfProgressCircular size="3xl" :value="loadingValue">
+          <text
+            class="font-medium fill-neutral-400"
+            text-anchor="middle"
+            alignment-baseline="central"
+            x="100%"
+            y="100%"
+          >
+            {{ loadingValue }}%
+          </text>
+        </SfProgressCircular>
+      </div>
+    </div>
+
+    <div v-if="loaded">
+      <SfInput v-model="fileName">
+        <template #suffix><SfIconClose @click="clearUploadedFile" /></template>
+      </SfInput>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { SfButton, SfInput, SfIconClose } from '@storefront-ui/vue';
+import { productPropertyGetters } from '@plentymarkets/shop-sdk';
+import { OrderPropertyInputProps } from './types';
+import { Ref, ref } from 'vue';
+const props = defineProps<OrderPropertyInputProps>();
+const productProperty = props.productProperty;
+const hasTooltip = props.hasTooltip;
+const value: Ref<Blob | null> = ref(null);
+
+const uploadForm: Ref<HTMLInputElement | null> = ref(null);
+const loading = ref(false);
+const loaded = ref(false);
+const loadingValue = ref(25);
+const fileName = ref('');
+
+const suportedFormats = [
+  'doc',
+  'docx',
+  'pdf',
+  'txt',
+  'obj',
+  'eps',
+  'ps',
+  'psd',
+  'sla',
+  'stp',
+  'step',
+  '3mf',
+  '3ds',
+  'max',
+  'jpg',
+  'png',
+  'svg',
+];
+
+const loadedFile: Ref<File | null> = ref(null);
+
+const handleFileUpload = (event: Event) => {
+  const target = event.target as HTMLInputElement;
+  if (target && target.files && target.files.length > 0) {
+    loadedFile.value = target.files[0];
+    const reader = new FileReader();
+    reader.addEventListener('load', (e) => {
+      if (loadedFile.value) {
+        fileName.value = loadedFile.value.name;
+        if (e && e.target && e.target.result) {
+          value.value = new Blob([e.target.result], { type: loadedFile.value.type });
+          loaded.value = true;
+        }
+      }
+    });
+    reader.readAsArrayBuffer(target.files[0]);
+  }
+};
```

Comment:
Should we check the type of the selected file?
Or we already send all acceptable types to the <input type="file" /> 
e.g.: https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/accept

Reply:
will do tomorow.

Reply:
The accept attribute doesn't validate the types of the selected files; it provides hints for browsers to guide users towards selecting the correct file types. It is still possible (in most cases) for users to toggle an option in the file chooser that makes it possible to override this and select any file they wish, and then choose incorrect file types.

Because of this, you should make sure that expected requirement is validated server-side.

Reply:
But i will check in the function

Reply:
Can we do both?

Reply:
not in scope of current task, but yes